game.py

```diff
@@ -39,7 +41,11 @@ def run(self) -> ChessPlayer:
         while self.terminated is False:
             player = self.player_white if self.board.turn is True else self.player_black
             try:
-                self.__make_a_move(player.get_next_move(self.board))
+                player_move = player.get_next_move(self.board)
+                if player_move == None:
+                    self.terminated = True
```

Comment:
This and the other null checks were potentially necessary because my player class doesn't check full end game state, just wins and draws. Without the two checks it goes into an infinite loop. If you have some ideas on how to mitigate this to avoid loops I'm all ears.